base/reduce.jl

```diff
@@ -281,15 +281,29 @@ mapreduce_impl(f, op, A::AbstractArrayOrBroadcasted, ifirst::Integer, ilast::Int
     mapreduce(f, op, itrs...; [init])
 
 Apply function `f` to each element(s) in `itrs`, and then reduce the result using the binary
-function `op`. If provided, `init` must be a neutral element for `op` that will be returned
-for empty collections. It is unspecified whether `init` is used for non-empty collections.
-In general, it will be necessary to provide `init` to work with empty collections.
+function `op`.
+
+The order of function evaluations and the associativity of the reduction is unspecified. Some
+implementations may reuse the return value of `f` for elements that appear multiple times in
+`itr`. Use [`mapfoldl`](@ref) or [`mapfoldr`](@ref) for
+strict left or right associativity and guaranteed invocation of `f` for every value.
+
+If provided, `init` serves as the return value for empty `itrs`. For non-empty iterators,
```

Comment:
I think we could clarify that `init` is applied directly to the reduction rather than the collection. I.e., we reduce with `init` rather than `f(init)`. But I don't have a clear and concise suggestion for the statement.

Reply:
Yeah, I did initially have a phrase in there about `init` not being transformed by `f` at one point. I agree it'd be nice to include... I just deleted it in the editing process because it was more cumbersome than I liked and I had phrased it in the negative, which always leaves a bad taste in my mouth for docs.